state-chain/test-utilities/src/rich_test_externalities.rs

```diff
@@ -261,11 +267,64 @@ where
 		)
 	}
 
+	/// Keeps executing pallet hooks until the given predicate returns true.
+	///
+	/// Preserves the context.
+	#[track_caller]
+	pub fn then_process_blocks_until<F: Fn(Ctx) -> bool>(mut self, predicate: F) -> Self {
```

Comment:
nit, predicates usually get `P` 
```suggestion
	pub fn then_process_blocks_until<P: Fn(Ctx) -> bool>(mut self, predicate: P) -> Self {
```